hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/container/ListSubcommand.java

```diff
@@ -105,12 +119,39 @@ public void execute(ScmClient scmClient) throws IOException {
           ReplicationType.fromProto(type),
           replication, new OzoneConfiguration());
     }
-    List<ContainerInfo> containerList =
-        scmClient.listContainer(startId, count, state, type, repConfig);
+
+    int maxCountAllowed = parent.getParent().getOzoneConf()
+        .getInt(ScmConfigKeys.HDDS_CONTAINER_LIST_MAX_COUNT,
+            ScmConfigKeys.HDDS_CONTAINER_LIST_MAX_COUNT_DEFAULT);
+    if (all) {
+      System.out.printf("Attempting to list all containers." +
+          " The total number of container might exceed" +
+          " the cluster's current limit of %s. The results will be capped at the" +
+          " maximum allowed count.%n", ScmConfigKeys.HDDS_CONTAINER_LIST_MAX_COUNT_DEFAULT);
```

Comment:
Should this and similar informational messages go to `System.err`?

Reply:
Yes, since these messages warn the user that not all data is being shown due to a restriction; it should be printed to `System.err`.